tools/src/bin/dependency_check.rs

```diff
@@ -0,0 +1,220 @@
+            Some(code) => bail!("`tar` command failed with code: {code}"),
+            None => bail!("`tar` command failed"),
+        },
+        Err(error) => return Err(anyhow::Error::from(error).context("`tar` command failed")),
+    }
+
+    Ok(())
+}
+
+fn download_ffmpeg(executable_dir: &Path) -> Result<()> {
+    let response = get(ffmpeg_url())
+        .with_context(|| format!("Failed to download libav libraries. URL: {}", ffmpeg_url()))?;
+    let content = response.bytes()?;
+
+    fs::write(executable_dir.join(FFMPEG_ARCHIVE_NAME), &content)
+        .with_context(|| "Failed to save downloaded libav libraries to file")?;
+    Ok(())
+}
+
+#[cfg(target_os = "linux")]
+fn check_ffmpeg() -> Result<bool> {
+    check_ffmpeg_command()
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg() -> Result<bool> {
+    let command_result = check_ffmpeg_command()?;
+    if !command_result {
+        info!("Checking if FFmpeg is installed as homebrew keg-only");
+        return check_ffmpeg_homebrew();
+    }
+    Ok(command_result)
+}
+
+fn check_ffmpeg_command() -> Result<bool> {
+    let ffmpeg_result = Command::new("ffmpeg").arg("-version").output();
+    match ffmpeg_result {
+        Ok(ffmpeg_output) => {
+            let ffmpeg_output = String::from_utf8(ffmpeg_output.stdout)?.trim().to_string();
+            match match_ffmpeg_version(&ffmpeg_output) {
+                Some(version) if version == required_ffmpeg_version() => Ok(true),
+                Some(version) => {
+                    warn!(
+                        installed_ffmpeg_version = version,
+                        required_ffmpeg_version = required_ffmpeg_version(),
+                        "Installed version doesn't match the required version."
+                    );
+                    Ok(false)
+                }
+                None => {
+                    warn!("Failed to parse FFmpeg version.");
+                    Ok(false)
+                }
+            }
+        }
+        Err(_) => {
+            warn!("Failed to run FFmpeg.");
+            Ok(false)
+        }
+    }
+}
+
+fn match_ffmpeg_version(ffmpeg_output: &str) -> Option<&str> {
+    // Matches ffmpeg version in the output of `ffmpeg -version` command, then returns the version.
+    // E.g. "ffmpeg version 8.0" (captures 8.0) or "ffmpeg version n7.1" (captures 7.1)
+    let re = Regex::new(r"(?m)^ffmpeg version \D*(\d+\.\d+)")
+        .expect("Failed to compile regular expression");
+    let caps = re.captures(ffmpeg_output);
+    match caps {
+        Some(caps) => caps.get(1).map(|cap| cap.as_str()),
+        None => {
+            warn!("Failed to parse FFmpeg version.");
+            None
+        }
+    }
+}
+
+#[cfg(target_os = "macos")]
+fn check_ffmpeg_homebrew() -> Result<bool> {
+    let required_ffmpeg_version_brew =
+        &required_ffmpeg_version()[..required_ffmpeg_version().find(".").unwrap_or(1)];
+    let brew_output = Command::new("brew").arg("list").output()?;
```

Comment:
What if there is no brew?
- Is the error clear?
- Can we provide better behavior? e.g. always download dependencies in that case? Would our binaries even work without homebrew?